pkg/builtins/ec_op.go

```diff
@@ -0,0 +1,262 @@
+	"github.com/lambdaclass/cairo-vm.go/pkg/vm/memory"
+	"math/big"
+	"reflect"
+)
+
+type EcOpBuiltinRunner struct {
+	included           bool
+	base               memory.Relocatable
+	cells_per_instance uint32
+	cache              map[memory.Relocatable]lambdaworks.Felt
+	n_input_cells      uint32
+	scalar_height      uint32
+}
+
+type EcPoint struct {
+	x uint
+	y uint
+}
+
+type PartialSum struct {
+	X lambdaworks.Felt
+	Y lambdaworks.Felt
+}
+
+type DoublePoint struct {
+	X lambdaworks.Felt
+	Y lambdaworks.Felt
+}
+
+type PartialSumB struct {
+	X big.Int
+	Y big.Int
+}
+
+type DoublePointB struct {
+	X big.Int
+	Y big.Int
+}
+
+const INPUT_CELLS_PER_EC_OP = 5
+const CELLS_PER_EC_OP = 7
+
+const PRIME = "0x800000000000011000000000000000000000000000000000000000000000001"
+
+func NewEcOpBuiltinRunner() *EcOpBuiltinRunner {
+	return &EcOpBuiltinRunner{
+		cells_per_instance: CELLS_PER_EC_OP,
+		n_input_cells:      INPUT_CELLS_PER_EC_OP,
+		cache:              make(map[memory.Relocatable]lambdaworks.Felt),
+		scalar_height:      256,
+	}
+}
+
+func (ec *EcOpBuiltinRunner) AddValidationRule(*memory.Memory) {}
+
+func (ec *EcOpBuiltinRunner) Base() memory.Relocatable {
+	return ec.base
+}
+
+func (ec *EcOpBuiltinRunner) Name() string {
+	return "ec_op"
+}
+
+func (ec *EcOpBuiltinRunner) InitializeSegments(segments *memory.MemorySegmentManager) {
+	ec.base = segments.AddSegment()
+}
+
+func (ec *EcOpBuiltinRunner) InitialStack() []memory.MaybeRelocatable {
+	if ec.included {
+		return []memory.MaybeRelocatable{*memory.NewMaybeRelocatableRelocatable(ec.base)}
+	} else {
+		return []memory.MaybeRelocatable{}
+	}
+}
+
+func (ec *EcOpBuiltinRunner) Include(include bool) {
+	ec.included = include
+}
+
+func (ec *EcOpBuiltinRunner) DeduceMemoryCell(address memory.Relocatable, segments *memory.Memory) (*memory.MaybeRelocatable, error) {
+	EC_POINT_INDICES := [3]EcPoint{{x: 0, y: 1}, {x: 2, y: 3}, {x: 5, y: 6}}
+	OUTPUT_INDICES := EC_POINT_INDICES[2]
+	alpha := lambdaworks.FeltOne()
+	beta_low := lambdaworks.FeltFromHex("0x609ad26c15c915c1f4cdfcb99cee9e89")
+	beta_high := lambdaworks.FeltFromHex("0x6f21413efbe40de150e596d72f7a8c5")
+	beta := (beta_high.Shl(128)).Add(beta_low)
+
+	index := address.Offset % uint(ec.cells_per_instance)
+
+	if index != OUTPUT_INDICES.x && index != OUTPUT_INDICES.y {
+		return nil, nil
+	}
+
+	instance := memory.NewRelocatable(address.SegmentIndex, address.Offset-index)
+	input_cells_per_ec_op := lambdaworks.FeltFromUint64(INPUT_CELLS_PER_EC_OP)
+	x_addr, err := instance.AddFelt(input_cells_per_ec_op)
+	if err != nil {
+		return nil, errors.New("Runner error, Expected Integer")
+	}
+
+	number := ec.cache[address]
+
+	if !reflect.DeepEqual(number, lambdaworks.Felt{}) {
+		return memory.NewMaybeRelocatableFelt(number), nil
+	}
+
+	//All input cells should be filled, and be integer values
+	//If an input cell is not filled, return None
+
+	input_cells := make([]lambdaworks.Felt, 0)
```

Comment:
```suggestion
	input_cells := make([]lambdaworks.Felt, INPUT_CELLS_PER_EC_OP)
```
We already know the size of the array, so we can avoid resizing with append

Reply:
i got a bug that when declaring an array when a fixed size, it fills it with 0s then when i append then it resizes  anyway leaving the 0s

Reply:
oh, then we can use `input_cells := make([]lambdaworks.Felt, 0, INPUT_CELLS_PER_EC_OP)` to set the len to 0 and capacity to INPUT_CELLS_PER_EC_OP